app/src/main/java/erevacation/com/userlistanko/homescreen/list/ListFragmentPresenter.kt

```diff
@@ -1,28 +1,55 @@
 package erevacation.com.userlistanko.homescreen.list
 
+import android.support.v4.app.FragmentManager
+import erevacation.com.userlistanko.R
 import erevacation.com.userlistanko.basic.arhitecture.ViperContract
+import erevacation.com.userlistanko.datamodel.ListDM
+import erevacation.com.userlistanko.homescreen.HomeActivity
+import erevacation.com.userlistanko.homescreen.profile.ProfileFragment
+import erevacation.com.userlistanko.usecase.ListUC
+import erevacation.com.userlistanko.usecase.ListUCContract
 import javax.inject.Inject
 
-class ListFragmentPresenter @Inject constructor() : ListContract.ListPresenter {
+class ListFragmentPresenter @Inject constructor(val listUC: ListUC) : ListContract.ListPresenter, ListUCContract.ListUCOut {
+
 
     private var view: ListContract.ListView? = null
     private val listAdapter = ListRecyclerAdapter(this)
+    private lateinit var fragmentManager: FragmentManager
 
     override fun viewAttach(view: ViperContract.View<*>) {
         this.view = view as ListContract.ListView
+        this.listUC.setListUCOut(this)
+        fragmentManager = ((view as ListFragment).context as HomeActivity).supportFragmentManager
+        listUC.getList()
         buildLayout()
+
     }
 
-    override fun viewDetach() {
+    override fun publishListResults(list: MutableList<ListDM>) {
+        listAdapter.updateList(list)
+    }
 
+    override fun viewDetach() {
+        this.view = null
```

Comment:
I think this method is missing, something like listUC.clearRequests()

Reply:
Correct. Great catch @djmoljkovic